hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java

```diff
@@ -87,19 +92,21 @@ public OzoneManagerStateMachine(OzoneManagerRatisServer ratisServer) {
     ThreadFactory build = new ThreadFactoryBuilder().setDaemon(true)
         .setNameFormat("OM StateMachine ApplyTransaction Thread - %d").build();
     this.executorService = HadoopExecutors.newSingleThreadExecutor(build);
+    this.installSnapshotExecutor = HadoopExecutors.newSingleThreadExecutor();
   }
 
   /**
    * Initializes the State Machine with the given server, group and storage.
    * TODO: Load the latest snapshot from the file system.
```

Comment:
This TODO looks a little worrying. Something we need to address now?

Reply:
Yes thanks for catching this. On startup, we should read the saved ratis snapshot index from disk. I will update the patch.

Reply:
Correction: On reloading state, we should not read the saved snaphsot index. Instead, we should updated the snapshot index on disk.
During normal startup, we already read the saved snapshot index.